### PR TITLE
Contrib: update to LanceDB contrib

### PR DESCRIPTION
**Pull request by zilto**
## For existing dataflows -- what has changed?
Added functions for vector and full text search and updated the README and the DAG accordingly.

## How I tested this
Used a notebook running all the nodes.

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Dataflow documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: contrib/hamilton/contrib/user/zilto/lancedb_vdb/README.md [hunk 1]
```diff
@@ -1,14 +1,17 @@
 # Purpose of this module
 
-This module implements simple operations to interact with LanceDB.
+This module implements vector and full-text search using LanceDB.
 
 # Configuration Options
 This module doesn't receive any configuration.
 
-## Inputs:
+## Required Inputs:
```
Comment (skrawcz) [inline contrib/hamilton/contrib/user/zilto/lancedb_vdb/README.md:8]: Why "required"?

**Comment by skrawcz** [comment]
Looking good otherwise -- just that one question.

File: contrib/hamilton/contrib/user/zilto/lancedb_vdb/README.md (hunk 1 above)
Comment (zilto) [reply contrib/hamilton/contrib/user/zilto/lancedb_vdb/README.md:8]: I thought that `Inputs` might imply that is an exhaustive list of all the inputs. Instead, this list includes that you will have to pass as input to use the module albeit not all concurrently

Comment (skrawcz) [reply contrib/hamilton/contrib/user/zilto/lancedb_vdb/README.md:8]: Yeah I think `Inputs` is fine. No need to for that extra qualification. Hamilton should complain anyway if something isn't satisfied.

Comment (skrawcz) [inline contrib/hamilton/contrib/user/zilto/lancedb_vdb/README.md:8]:
```suggestion
## Inputs:
```

**Review by skrawcz — APPROVED** [review]

**Pull request merged**